### ISSUE TITLE
Publish filtering option misses the count of entries

### ISSUE THREAD

**Issue by michael-e** [symphonycms/symphony-2#2129]
The new publish filtering is cool, but it doesn't display the count of entries. It would be useful to know

* how many entries are there if no filters are active
* how many entries have been found with the current filter(s)

**Comment by michael-e** [comment]
This issue is not solved (at least not in Symphony 2.6.0-beta.2).

**Comment by michael-e** [comment]
Ah, I see now: Filtered results get a count, but there is no count for an unfiltered list of entries. :-(

**Comment by nitriques** [comment]
I'll try to have a look at this

**Comment by michael-e** [comment]
Great, thanks!

**Comment by nitriques** [comment]
@michael-e  It's simply implemented that way

See https://github.com/symphonycms/symphony-2/blob/integration/symphony/content/content.publish.php#L497-L500

Would the solution be to create an `else` statement with Total entries: x ?

**Comment by michael-e** [comment]
Will test tomorrow, I am on my way to bed.

**Comment by nitriques** [comment]
@michael-e Nite nite.

**Comment by nilshoerrmann** [comment]
> Ah, I see now: Filtered results get a count, but there is no count for an unfiltered list of entries. :-(

Michael, would you like to always see a count or only if the filtering drawer is open?

**Comment by michael-e** [comment]
I would prefer to always see a count. It's very useful, IMHO.

**Comment by andrewminton** [comment]
+1

**Comment by michael-e** [comment]
For a quick test I hacked:

```php
else {
    $filter_stats = new XMLElement('p', '<span>' . __('(%d)', array(EntryManager::fetchCount($section_id))) . '</span>', array('class' => 'inactive'));
    $this->Breadcrumbs->appendChild($filter_stats);
}
```

right after the lines [mentioned by @nitriques](https://github.com/symphonycms/symphony-2/issues/2129#issuecomment-70181505), and it works fine for me.

To be consistent it only generates a number in brackets, so I am not sure if we should pass this to translation.

**Comment by nilshoerrmann** [comment]
If we will always display a count, I'd suggest to just add the count without brackets:

- 123 entries
- 12 of 123 entries

**Comment by michael-e** [comment]
Currently the filtered count is like this:

* – filtered (12 of 123)

I added:

* (123)

So we have to change the filtered count as well, so we get:

* (12 of 123 entries)
* (123)

Or would you prefer a dash instead of brackets?

**Comment by nilshoerrmann** [comment]
The dash is used in other places when we append something to the title (section index etc.).

**Comment by michael-e** [comment]
One way or another we must remove the word "filtered" to provide a consistent experience, right? So we'd have:

* `– 12 of 123 entries`
* `– 123 entries`

Or the word "filtered" gets moved to the end:

* `– 12 of 123 entries (filtered)`
* `– 123 entries`

Which dash is it, btw?

**Comment by nilshoerrmann** [comment]
> Which dash is it, btw? 

It should be an m-dash but we could just decide for something else. There are currently two kinds of additional information appended to headlines:

- "filtered (x of y)" and
- "drag to reorder" (added by the JavaScript, I think)

We should just make sure that the typesetting is identical.

**Comment by michael-e** [comment]
So what do you suggest?

* `– filtered (12 of 123 entries)`
* `– 123 entries`

Looks rather inconsistent, doesn't it? I would prefer

* `– 12 of 123 entries`
* `– 123 entries`

**Comment by bernardodiasc** [comment]
The count of entries is useful with and without filtering IMHO. Suggestion:

- `– 12 of 123 entries (filtered)`
- `– 12 of 123 entries`

**Comment by michael-e** [comment]
Without filtering, why should it be `12 of 123 entries`? It's simply `123 entries` then.

**Comment by bernardodiasc** [comment]
There is only 2 ways to know the count of entries in a page not filtered: knowing by the config.php or counting manually... I think its common to have the number of rows informed somewhere.

**Comment by michael-e** [comment]
You are mixing up things then. Symphony displays the overall number of filtered entries, not the number of visible rows. The issue discussed here is not about the number of visible rows.

**Comment by bernardodiasc** [comment]
alright. I imagined that you are talking about the label, not only for filters. without filter doesn't show the total entries nor rows anywhere.

**Comment by nilshoerrmann** [comment]
> without filter doesn't show the total entries nor rows anywhere.

The request was to change this.

**Comment by michael-e** [comment]
@nilshoerrmann, once we can agree on the text to be displayed, we could prepare a pull request for this. My suggestion is:

* `123 entries` (without filtering)
* `12 of 123 entries (filtered)` (if filtering is applied)

**Comment by nilshoerrmann** [comment]
I'm fine with that.

**Comment by michael-e** [comment]
So I will try my best.

**Comment by michael-e** [comment]
Ah, I see that [translations](https://github.com/hananils/lang_german/blob/master/lang/lang.de.php#L1770) would have to change as well. How is the right way to deal with that?

**Comment by nilshoerrmann** [comment]
Just change the the text in Symphony and make use of the `__()` function.
Using Localisation Manager, the translations can be easily updated.

**Comment by michael-e** [comment]
OK, will do.

**Comment by nitriques** [comment]
Can be closed since #2333 has been merged ?
